clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1184,8 +1184,9 @@ public void commitSync() {
      *             this function is called
      * @throws org.apache.kafka.common.errors.AuthorizationException if not authorized to the topic or to the
      *             configured groupId
+     * @throws java.lang.IllegalArgumentException if the committed offset is negative
      * @throws org.apache.kafka.common.KafkaException for any other unrecoverable errors (e.g. if offset metadata
-     *             is too large or if the committed offset is invalid).
+     *             is too large or if the topic does not exist).
```

Comment:
What is the offset commit is positive and invalid? cc @hachikuji

Reply:
You mean if the offset is out of range? I'm not sure we have a good way to check this at the moment. It can't be done on the coordinator because we don't know what the valid offsets are for each topic partition, so that leaves the client where the check may end up stale anyway.

By the way, there are a couple `commitSync` overloads that may need to be updated as well.

Reply:
Yes offsets are not checked at all on the broker side, allowing to commit any value. Also the negativity check is only performed client side, so it might not be done by all 3rd party clients. 
Do you think it's worth creating a small KIP to move this validation on the broker side ?

Anyway, I've updated the other `commitSync` override.

Reply:
Hmm.. I don't think it would necessarily require a KIP to check for negative offsets on the broker. I guess it depends what error code we return. In any case, we'd probably still want the validation on the client side.

Reply:
I'm happy for this to be merged if @hachikuji is happy fwiw.